lib/orbf/rules_engine/fetch_data/resolve_arguments.rb

```diff
@@ -26,22 +27,27 @@ def call
 
       def build_package_arguments
         project.packages.each_with_object({}) do |package, hash|
-          next unless match_group?(package)
+
+          orgunits = decorate_with_facts(build_orgunit_resolver(package, pyramid, main_orgunit).call)
 
           hash[package] = PackageArguments.with(
             periods:          PeriodsResolver.new(package, invoicing_period).call,
             orgunits:         OrgUnits.new(
               package:  package,
-              orgunits: decorate_with_facts(OrgunitsResolver.new(package, pyramid, main_orgunit).call)
+              orgunits: orgunits
             ),
             datasets_ext_ids: DatasetsResolver.dataset_extids(package),
             package:          package
           )
         end
       end
 
-      def match_group?(package)
-        (package.main_org_unit_group_ext_ids & main_orgunit.group_ext_ids).any?
+      def build_orgunit_resolver(package, pyramid, main_orgunit)
+        if @contract_service
+          ContractOrgunitsResolver.new(package, pyramid, main_orgunit, @contract_service, invoicing_period)
+        else
+          OrgunitsResolver.new(package, pyramid, main_orgunit)
```

Comment:
Thinking out loud here, would it make sence to move the `if @contract_service` inside the `OrgunitsResolver` and let that one use the `ContractOrgunitsResolver` when needed?

Reply:
I wanted to keep things "seperated". and the existing OrgunitsResolver is allready complex.
and wanted to avoid passing "unused" params like invoicing_period and 

I may propose : 
- rename OrgunitsResolver to GroupsOrgunitsResolver 
- have a fresh OrgunitsResolver doing the switch between GroupsOrgunitsResolver and ContractOrgunitsResolver